_examples/how-to-use-planetscale-in-your-serverless-app.md

```diff
@@ -0,0 +1,301 @@
+
+> Make sure you have [pscale CLI](https://docs.planetscale.com/reference/planetscale-environment-setup) installed locally
+
+Sign in to your account
+
+```bash
+pscale auth login
+```
+
+You'll be taken to a screen in the browser where you'll be asked to confirm the code displayed in your terminal. If the confirmation codes match, click the "Confirm code" button in your browser.
+
+You should receive the message "Successfully logged in" in your terminal. You can now close the confirmation page in the browser and proceed in the terminal.
+
+### Creating a database
+
+Now that you have signed into PlanetScale using the CLI, you can create a database.
+
+Run the following command to create a database named `demo`:
+
+```bash
+pscale database create demo
+```
+
+### Add a schema to your database
+
+To add a schema to your database, you will need to connect to MySQL using the pscale shell.
+
+Run the following command:
+
+```bash
+pscale shell demo main
+```
+
+You are now connected to your main branch and can run MySQL queries against it.
+
+Create a table by running the following command:
+
+```sql
+CREATE TABLE IF NOT EXISTS counter (counter VARCHAR(255) PRIMARY KEY, tally INT);
+```
+
+You can confirm that the table has been added by running the following command:
+
+```sql
+DESCRIBE counter;
+```
+
+Insert data into your database
+Now that you have created a counter table, you can insert data into that table.
+
+Run the following command to add a hits counter to your table.
+
+```sql
+INSERT INTO counter (counter, tally) VALUES ('hits', 0) ON DUPLICATE KEY UPDATE tally = 0;
+```
+
+![terminal output](/assets/examples/planetscale/terminal-output.png)
+
+We will be using the PlanetScale API to provision a TLS certificate, and then connects to the database. It uses Service Tokens for authentication, so you'll need to create one for the app:
+
+```bash
+$ pscale service-token create
+  NAME           TOKEN
+ -------------- ------------------------------------------
+  nyprhd2z6bd3   [REDACTED]
+
+$ pscale service-token add-access nyprhd2z6bd3 connect_branch --database demo
+  DATABASE   ACCESSES
+ ---------- ---------------------------
+  demo       connect_branch
+```
+
+Add the above values in the `.env` file in the root
+
+```
+PLANETSCALE_TOKEN='<TOKEN_FROM_TERMINAL>'
+PLANETSCALE_TOKEN_NAME='nyprhd2z6bd3'
+PLANETSCALE_ORG='<ORG_NAME>'
+PLANETSCALE_DB='demo'
+```
+
+> _NOTE_ : The names of the env variables should not be changed
+
+## Setting up the API
+
+Now let's add the API.
+
+{%change%} Replace the code in `stacks/MyStack.js` with below.
+
+```js
+import * as sst from "@serverless-stack/resources";
+
+export default class MyStack extends sst.Stack {
+  constructor(scope, id, props) {
+    super(scope, id, props);
+
+    // Create a HTTP API
+    const api = new sst.Api(this, "Api", {
+      routes: {
+        "POST /": {
+          function: {
+            handler: "src/lambda.handler",
+            environment: {
```

Comment:
Let's move `environment` to `defaultFunctionProps`. And have the route look like:
```
"POST /": "src/lambda.handler",
```
Looks cleaner this way.